File: backend/src/models/Unit.js
const mongoose = require('mongoose');

const unitSchema = new mongoose.Schema({
  courseId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Course',
    required: true,
  },
  title: { type: String, required: true, trim: true },
  description: { type: String, default: '' },
  order: { type: Number, required: true },
}, {
  timestamps: true,
  toJSON: {
    transform(doc, ret) {
      ret.id = ret._id;
      delete ret._id;
      delete ret.__v;
      return ret;
    },
  },
});

unitSchema.index({ courseId: 1, order: 1 });

module.exports = mongoose.model('Unit', unitSchema);

File: mobile/lib/domain/entities/unit.dart
import 'package:equatable/equatable.dart';
import 'lesson.dart';

class Unit extends Equatable {
  final String id;
  final String courseId;
  final String title;
  final String description;
  final int order;
  final List<Lesson> lessons;

  const Unit({
    required this.id,
    required this.courseId,
    required this.title,
    required this.description,
    required this.order,
    this.lessons = const [],
  });

  @override
  List<Object?> get props => [id, courseId, title, description, order, lessons];
}
